lib/beaker/dsl/helpers/web_helpers.rb

```diff
@@ -52,9 +53,17 @@ def fetch_http_file(base_url, file_name, dst_dir)
           else
             logger.notify "Fetching: #{src}"
             logger.notify "  and saving to #{dst}"
-            open(src, :allow_redirections => :all) do |remote|
-              File.open(dst, "w") do |file|
-                FileUtils.copy_stream(remote, file)
+            begin
+              open(src, :allow_redirections => :all) do |remote|
+                File.open(dst, "w") do |file|
+                  FileUtils.copy_stream(remote, file)
+                end
+              end
+            rescue OpenURI::HTTPError => e
+              if e.message =~ /404.*/
+                raise "Failed to fetch_remote_file '#{src}' (exit code #{$?}"
```

Comment:
Whoa, I'm having trouble figuring out what `$?` is going to get.  Typically I see `exit_code` being called on something to get the exit code.   This looks like bash but I don't see any bash executed nearby?

Also, the message is missing the trailing `)` after the exit code.